core/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/SubmissionComponent.scala

```diff
@@ -151,11 +151,25 @@ trait SubmissionComponent {
         })
       )
 
-    def listWithSubmitter(workspaceContext: Workspace): ReadWriteAction[Seq[SubmissionListResponse]] = {
+    def listWithSubmitter(workspaceContext: Workspace,
+                          startDateOpt: Option[DateTime],
+                          endDateOpt: Option[DateTime]
+    ): ReadWriteAction[Seq[SubmissionListResponse]] = {
+      val baseQuery = findByWorkspaceId(workspaceContext.workspaceIdAsUUID)
+      val filteredQuery = (startDateOpt, endDateOpt) match {
+        case (Some(startDate), Some(endDate)) =>
+          val start = new Timestamp(startDate.withTimeAtStartOfDay.getMillis)
+          val end = new Timestamp(endDate.plusDays(1).withTimeAtStartOfDay.getMillis - 1)
+          baseQuery.filter(sub =>
+            sub.submissionDate >= start && sub.submissionDate <= end
+          )
+        case _ =>
+          baseQuery
```

Comment:
Could you apply the start filter and the end filter independently? That is, if user only supplies one of the query params, that query param will still take effect?

In that case, "last 30 days" would only need to specify the start date, and can omit the end date.

Reply:
Yes! I just made that change